### DISCUSSION TITLE
Probleme beim Binding mit String und Numerischen Datentypen

### DISCUSSION THREAD

**Discussion by Erwin40** [Zefau/ioBroker.jarvis#2727] [General]
Hallo zusammen.

_Aktuell 3.2.0-beta.49_
Ich bekomme es einfach nicht hin. Nachfogendes Binding im Geräte-DP im Feld **Icon** eingetragen:
{
  "CLOSE": "mdi:window-shutter-auto",
  "OPEN": "data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjQiIGhlaWdodD0iMjQiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyI+CgogPGc+CiAgPHRpdGxlPkxheWVyIDE8L3RpdGxlPgogIDxwYXRoIGlkPSJzdmdfMSIgZD0ibTIwLDE0bC0yLDBsLTMuMiw5bDEuOSwwbDAuNywtMmwzLjIsMGwwLjcsMmwxLjksMGwtMy4yLC05em0tMi4yLDUuN2wxLjIsLTMuN2wxLjIsMy43bC0yLjQsMHptLTEwLjgsLTEwLjdsOCwwbDAsMmwtOCwwbDAsLTJ6bTksLTFsLTEwLDBsMCwxMmwtMiwwbDAsLTEybC0yLDBsMCwtNGwxOCwwbDAsNGwtMiwwbDAsNGwtMS40LDBsLTAuNSwxLjNsLTAuMSwwLjRsMCwtNS43eiIgZmlsbD0iY3VycmVudENvbG9yIi8+CiA8L2c+Cjwvc3ZnPg==",
  "'{val}'=='STOP' && {0_userdata.0.jarvis.Rolladen.Rolladen_position_All}'<=49": "mdi:temperature-chevron-down",
  "'{val}'=='STOP' && {0_userdata.0.jarvis.Rolladen.Rolladen_position_All}'<=100": "mdi:temperature-chevron-up"
}

Es hieß ja, bei String DP müssen beide (Name und Value) in Hochkommas.
Der zweite DP, der als Binding drin ist, ist aber numerisch.
Wenn ich das so eintrage knallt es richtig:
![grafik](https://github.com/user-attachments/assets/5cab8718-e8af-49e3-82e4-08b994e9719d)

Setze ich auch das Binding, also den Teil rechts vom && in Hochkommas, kommt zwar die Warnung nicht, aber im Console-Log ist trotzdem eine Fehlermeldung:
`15:51:28.841 StateList be31acad-6bb4-40ed-a9cf-24cd981d8545 Alle Rolläden position 
Object { val: 11, ack: false, ts: 1722345664039, q: 0, from: "system.adapter.jarvis.0", user: "system.user.admin", lc: 1722345664039, id: "0_userdata.0.jarvis.Rolladen.Rolladen_position_All", stateKey: "position", revision: "57fa0212-d050-4fa1-a872-9317f47b6531", … }
 val}'=='STOP' && '{0_userdata.0.jarvis.Rolladen.Rolladen_position_All is not a valid state (in getState) index-cc169059.js:1698:9661
`

Kann mir hier bitte jemand auf die Sprünge helfen? :-)

**Comment by mcuiobroker** [comment]
Lass mal die ' bei {val} weg
```
"{val}=='STOP' && {0_userdata.0.jarvis.Rolladen.Rolladen_position_All}<=49": "mdi:temperature-chevron-down",
"{val}=='STOP' && {0_userdata.0.jarvis.Rolladen.Rolladen_position_All}<=100": "mdi:temperature-chevron-up"
```
Ein Hochkomma zu viel
![image](https://github.com/user-attachments/assets/2a3368b7-3006-4748-ab44-2279660da98b)

**Comment by Erwin40** [comment]
Leider immer noch nicht.
![grafik](https://github.com/user-attachments/assets/dde0383c-08ae-4f55-a450-696cb5cb5104)

So habe ich es auch versucht, da kommt die gleiche Meldung:
"{val}='STOP' && {0_userdata.0.jarvis.Rolladen.Rolladen_position_All}<49": "mdi:temperature-chevron-down",
"{val}='STOP' && {0_userdata.0.jarvis.Rolladen.Rolladen_position_All}<100": "mdi:temperature-chevron-up"

Lustigerweise funktioniert das hier wunderbar ohne Meldungen im Log:
{
  "{val}='STOP'": "mdi:temperature-chevron-down",
  "{0_userdata.0.jarvis.Rolladen.Rolladen_position_All_C}<100": "mdi:temperature-chevron-up"
}

Das ist der Geräte-DP in dem das Icon gesetzt werden soll, Inhalt ("STOP"):
  "common": {
    "name": "Rolladen_state All",
    "desc": "Manuell erzeugt",
    "role": "switch",
    "states": {
      "OPEN": "OPEN",
      "CLOSE": "CLOSE",
      "STOP": "STOP",
      " ": "nix"
    },
    "type": "string",
    "read": true,
    "write": true,
    "def": ""
  },

Das ist der ioBroker-DP {0_userdata.0.jarvis.Rolladen.Rolladen_position_All}  Inhalt (**11**):
  "common": {
    "name": "Rolladen_position All",
    "desc": "Manuell erzeugt",
    "role": "state",
    "type": "number",
    "read": true,
    "write": true,
    "def": 0,
    "unit": "%",
    "min": 0,
    "max": 100
  },

So langsam bin ich am verweifeln *schnief*

**Comment by mcuiobroker** [comment]
Dann muss sich @zefau das genauer anschauen. Evtl. als Issue-Bug neu reinstellen.

**Comment by Zefau** [comment]
Im Screenshot siehst du, dass `{val}` auf `null` aufgelöst wird:

![353566734-dde0383c-08ae-4f55-a450-696cb5cb5104](https://github.com/user-attachments/assets/2ef59a81-f896-4d9a-8933-92597413e630)

```
'null' == 'STOP'
```

Da stimmt also was mit deinem Datenpunkt nicht.

**Comment by Zefau** [comment]
https://github.com/Zefau/ioBroker.jarvis/issues/2729
